module_3/gifts/controller/src/main/java/com/epam/esm/controller/StatisticController.java

```diff
@@ -0,0 +1,40 @@
+package com.epam.esm.controller;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.epam.esm.dto.TagDto;
+import com.epam.esm.service.TagService;
+
+/**
+ * Controller for working with statistics.
+ * 
+ */
+@RestController
+@RequestMapping("/api/v1/statistics")
+public class StatisticController {
+
+	private TagService tagService;
+
+	@Autowired
+	public StatisticController(TagService tagService) {
+		this.tagService = tagService;
+	}
+
+	/**
+	 * Selects the most widely used tag of a user with the highest cost of all
+	 * orders.
+	 * 
+	 * @return the most widely used tag of a user with the highest cost of all
+	 *         orders
+	 */
+	@GetMapping("/popular/tag")
```

Comment:
@LizavetaSinitsyna In this case I thing would be greate to start from tag. `/tags/popular.` It allows you to use other mapping. For example:` /certificates/popular `

Reply:
@AliaksandrLiulis, if I change the url in such way ("/statistics/tags/popular) would it be better to move this method to TagController and remove the separate StatisticController? Because all urls for tags are formed in such way "/tags/*", so may be there is no need for separate controller?